src/gt4py/cartesian/frontend/nodes.py

```diff
@@ -168,9 +168,10 @@ class Location(Node):
     scope = attribute(of=str, default="<source>")
 
     @classmethod
-    def from_ast_node(cls, ast_node, scope="<source>"):
+    def from_ast_node(cls, ast_node, scope: str | None = None):
```

Comment:
Just a comment: since the default value is immutable, there is nothing wrong with assigning it here and it is more explicit.

Additionally, if you _reaaally_ want to use the same default as in the class init, you could do:
```suggestion
    def from_ast_node(cls, ast_node, scope: str = scope._default):
```